### ISSUE TITLE
Database loading and checking options for currently supported databases

### ISSUE THREAD

**Issue by dhrogers** [cinemascience/cinema_scope#20]
Current implementation only supports a limited set of Cinema databases. Reader/loader should report on unsupported databases, and allow 1) recovery, 2) definition of expected columns (phi, theta, time, FILE, etc.)

Command line check of database should also be supported.

**Comment by dhrogers** [comment]
First iteration of this is done. If a database is loaded and doesn't have the expected columns ('phi', 'theta', 'FILE') a dialog pops up allowing the user to choose columns.

TBD:
1. if the selected artifact is not a supported file, the system should report that. Currently, the artifacts will fail to load, but there is no feedback.

**Comment by terryturton** [comment]
Adding to this issue would be a check on column names to flag non-C++ compliant names e.g. a space or any non-underscore symbols.

**Comment by dhrogers** [comment]
@terryturton : Agreed, there should be more detail here, but just to clarify we are trying to achieve two things:

1. Compliance with the current spec,
2. Constraints from the application.

There is work to do on #1, but mainly for things like checking to make sure that all values in a column are the same type, etc.

I believe that the specific things you address in your comment above are covered in the Cinema Spec.

**Comment by terryturton** [comment]
We might have to update the specifications.  Cinema spec D:
_The file follows the specification rfc4180 [4], governing how fields are defined._

But the rfc4180 specification for CSV files states:
_Spaces are considered part of a field and should not be ignored_

CSV fields are more generously defined than C++ variables.
